opentracing/span.py

```diff
@@ -209,17 +212,26 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         """Ends context manager and calls finish() on the :class:`Span`.
 
-        If exception has occurred during execution, it is automatically added
-        as a tag to the :class:`Span`.
+        If exception has occurred during execution, it is automatically logged
+        and added as a tag. :attr:`~operation.ext.tags.ERROR` will also be set
+        to `True`.
         """
-        if exc_type:
-            self.log_kv({
-                'python.exception.type': exc_type,
-                'python.exception.val': exc_val,
-                'python.exception.tb': exc_tb,
-                })
+        self._on_error(exc_type, exc_val, exc_tb)
         self.finish()
 
+    def _on_error(self, exc_type, exc_val, exc_tb):
+        if not exc_val:
+            return
+
+        self.set_tag(tags.ERROR, True)
+        self.log_kv({
+            logs.EVENT: tags.ERROR,
+            logs.ERROR_OBJECT: exc_val,
+            logs.ERROR_KIND: exc_type.__name__,
+            logs.MESSAGE: str(exc_val),
+            logs.STACK: traceback.format_exc(),
```

Comment:
I have a slight preference to passing all 3 values above as is, without any additional work. If implementation wants to call `.format_exc()`, it can do so, but doing it explicitly in the API introduces a lot of unavoidable overhead, especially when the trace is not sampled.

Reply:
In that case, we *might* need to use different log names, as per the spec, strings are expected. What's your take on this? Or simply put the objects there directly, anyway? (Personally I have mixed feelings about this, but I'd guess the latter option would be better ;) )

Reply:
I think the spec is wrong about that. Forcing to-string prematurely is the biggest no-no of monitoring APIs. If we take Jaeger tracer, it will do to-string anyway, but only if the trace is actually sampled and the conversion is necessary. In some cases to-string is not even appropriate, e.g. if a tracing backend supports stack traces via proper structure, then I would much rather get the language-specific date structure and pack it more efficiently than to-string.

Reply:
All right, lets do that. I just updated the PR ;)